Feat/brooke/settings

brooke-k opened the pull request:
- Created a settings page, with unique colour and based on the modals used for car, navigation, music, and climate.

- Created a button in the top right corner of the home page to access settings, done in similar styling to the buttons in the bottom toolbar. 

- Added dark and light mode functionality, which can be toggled manually between dark and light mode using the toggle switch found on the settings page

- Added automatic detection of the user's local dark/light mode preference, and will automatically activate the preferred colour mode. If the device being used does not have a preferred colour mode, the UI will default to using the dark theme. Manually toggling dark and light mode with the toggle in settings will override automatic detection, so the user can change the colour mode if they do not want the one that is automatically preferred by their device. 

- Modified colours in the variables.scss document to better suit dark/light mode variability, and allow for all visual components to be able to be fully usable in both themes. The Ionic theme colours "secondary" and "tertiary," were modified from different shades of blue that came as default with Ionic, to two different shades of grey, to allow for greater variability and use in different visual components throughout the UI. 

- Modified colours and variable.scss references in numerous files throughout the UI to allow for all visual components to be able to flip between dark and light mode, while still being legible and aesthetically pleasing, as well as unifying the entirety of the UI in regards to colour. Both .tsx and .scss files were modified to properly accommodate this. All colour-changing features of the climate page were not affected, and continue to maintain the same functionality as was present before the surrounding colour changes. 

- No features of the UI, outside of the created settings page, were modified beyond colour modifications.

brooke-k committed:
Add button to access the settings page to the home page

- Added a floating action button to the top right of the home page to access the settings page. The settings page pops up in a modal, identically to the other car, navigation, climate, and music pages and can also be closed with either the 'X' button in the top left corner, or by clicking outside of the modal. 
There is no content in the settings modal; it is an empty page.
- Created a new indigo-blue colour for use of the toolbar at the top of the settings modal. The colour has been added to the variables.scss document.

brooke-k committed:
Added dark/light mode toggle to settings to allow user to change the
color scheme. Modified the variables.scss file to contain color schemes
for both automatic rendering, based on the user's system, and a for
maually toggling dark mode on and off. When the UI is started up, the
default color scheme is whatever the user's device is set to.

Added settings page and dark/light mode optionality.

- Created a basic settings page, with a functional toggle for changing between the dark and light-mode colour schemes for the UI. The variables.scss document has had most colours changed, to accommodate the light and dark-mode functionality and make sure that visibility is maintained with both colour themes. 
- All files that either contained previous variable.scss references, or contained content that would be affected by the dark and light-mode toggling had the colour methods adjusted to make toggling work fully. 
- The climate page's colour functionality for direction buttons, the top bar, or any other component using the colorCheck hook has not been touched and continues to work as it did prior to the dark/light adjustment. 
- Details about what each colour needed for the theme should be referenced as from this point forward are outlined in the variables.scss document.

rjb75 commented on src/components/SettingsFab.tsx:
```diff
@@ -0,0 +1,35 @@
+    const page = useSelector(selectRouter);
+
+    const handleClick = (button: Pages) => {
+        console.log(button);
```
@brooke-k  do we need to keep this `console.log()` here?

brooke-k replied:
Removed it in the latest commit. Not sure why it was there, but it wasn't doing anything and I double-checked to make sure that nothing broke when I removed it - nothing did, so it's gone now.

rjb75 commented on src/pages/Settings.tsx:
```diff
@@ -1,23 +1,77 @@
+                </IonList>
             </IonContent>
+
+            {/* <IonContent className="ModalContent" fullscreen={false}>
```
@brooke-k did you want to keep this block in for reference?

brooke-k replied:
This was just me playing around with methods for getting the settings page set up in a possibly better formatted matter. This block, however, is completely non-functioning and I honestly just forgot to get rid of it. It's been removed in the latest commit.

rjb75 reviewed (changes requested):
Looks good, I really like the light mode! Just saw a couple things that we could remove before merging.

brooke-k committed:
Removed 'console.log(button)' that was unnecessary from SettingsFab.tsx.

Removed commented out code from Settings.tsx. It was unnecessary to
save, and was non-functioning.
